exampleCourse/questions/demo/annotated/engCircuit/server.py

```diff
@@ -1,40 +1,77 @@
 import random
+import io
 
+import matplotlib.pyplot as plt
+from schemdraw import Drawing
+import schemdraw.elements as elm
+
+def file(data):
+    if data["filename"] == "figure.png":
+        drawing = Drawing()
+        R1 = str(data["params"]["R1"]) + " $\\Omega$"
+        R2 = str(data["params"]["R2"]) + " $\\Omega$"
+        R3 = str(data["params"]["R3"]) + " $\\Omega$"
+        Vt = str(data["params"]["Vt"]) + "V"
+        if data["params"]["ask"] == "equivalent resistance $R_T$":
+            ## parallel resistors
+            drawing.push()
+            drawing += elm.Line()
+            drawing.push()
+            drawing += elm.Resistor().down().label(R1)
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing.push()
+            drawing += elm.Resistor().down().label(R2)
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing += elm.Resistor().down().label(R3)
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing.pop()
+            drawing += elm.BatteryCell().down().label(Vt,'bottom')
+        else:
+            ## Series resistors
+            drawing.push()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().down()
+            drawing += elm.Resistor().left().label(R3)
+            drawing += elm.Resistor().left().label(R2)
+            drawing += elm.Resistor().left().label(R1)
+            drawing.pop()
+            drawing += elm.BatteryCell().down().label(Vt,'bottom')
+        drawing.draw(show=False)
+        buf = io.BytesIO()                       
+        plt.savefig(buf, format="png")  
+        return buf
 
 def generate(data):
     ask = ["equivalent resistance $R_T$", "current from the power supply $I_T$"]
     which = random.choice([0, 1])
     data["params"]["ask"] = ask[which]
-
     label = ["$R_T$", "$I_T$"]
     data["params"]["lab"] = label[which]
 
-    unit = ["$\\Omega$", "A"]
-    data["params"]["unit"] = unit[which]
-
     Vt = random.randint(100, 200)
     data["params"]["Vt"] = Vt
-
```

Comment:
This code is arguably much more readable with the newlines - any reason in particular you removed them?

Reply:
No reason as such, will add them back in the PR. Thanks for feedback 👍